Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -893,23 +921,31 @@ WarpXOpenPMDPlot::SetupPos (
     openPMD::ParticleSpecies& currSpecies,
     const unsigned long long& np,
     amrex::ParticleReal const charge,
-    amrex::ParticleReal const mass) const
+    amrex::ParticleReal const mass,
+    bool const isBTD)
 {
-  auto const realType = openPMD::Dataset(openPMD::determineDatatype<amrex::ParticleReal>(), {np});
-  auto const idType = openPMD::Dataset(openPMD::determineDatatype< uint64_t >(), {np});
+  std::string options = "{}";
+  if (isBTD) options = "{ \"resizable\": true }";
+  auto realType= openPMD::Dataset(openPMD::determineDatatype<amrex::ParticleReal>(), {np}, options);
+  auto idType = openPMD::Dataset(openPMD::determineDatatype< uint64_t >(), {np}, options);
 
   auto const positionComponents = detail::getParticlePositionComponentLabels();
   for( auto const& comp : positionComponents ) {
       currSpecies["positionOffset"][comp].resetDataset( realType );
-      currSpecies["positionOffset"][comp].makeConstant( 0. );
       currSpecies["position"][comp].resetDataset( realType );
   }
 
   auto const scalar = openPMD::RecordComponent::SCALAR;
   currSpecies["id"][scalar].resetDataset( idType );
   currSpecies["charge"][scalar].resetDataset( realType );
-  currSpecies["charge"][scalar].makeConstant( charge );
   currSpecies["mass"][scalar].resetDataset( realType );
+
+  if (isBTD and m_ParticleSetUp == false) return;
```

Comment:
Shouldn't this read 
```suggestion
  if (isBTD and m_ParticleSetUp == true) return;
```
so that it returns if the particle are already set up?

Reply:
oh!!! thanks for catching this!!!

Reply:
I see the confusion here.
`m_ParticleSetUp` is set to true when we want to set the meta-data and if its false, then we return.
This logic is fine, and consistent with how we set up the particles.
Either way, I will test it again and see

Reply:
Discussed: let's rename to `m_doParticleSetup